### ISSUE TITLE
force recheck / check not seeing file as 100% present

### ISSUE THREAD

**Issue by themusicgod1** [Tribler/tribler#3499]
#### Please, fill all relevant items: ####

- [x] I have read CONTRIBUTING.rst

- [x] I have tried with the latest pre-release version (next) and I still can reproduce the issue.

##### Tribler version/branch+revision: #####

both 7.0.1 and next + 0f36719e6

##### Operating system and version: #####

ubuntu artful / 17.10

##### Steps to reproduce the behavior: #####

1) download torrent from https://archive.org/download/NINETEENEIGHTY-FOUR1984ByGeorgeOrwellPDFAudioBook
2) notice one of the files is 1984.pdf
3) stop torrent
4) download that 1984.pdf from archive.org & place it in download directory
5) force recheck

##### Expected behavior: #####

6) file 1984.pdf should be  at 100% in tribler

##### Actual behavior: #####

6) file 1984.pdf is still at 0% in tribler

##### Relevant log file output: #####

????

WARNING 1520539682.06       community:500   message from {10.0.1.10:7759 81.92.222.206:7759} cannot be found in the request cache, skipping it
WARNING 1520539687.15       community:500   message from {192.168.1.3:7759 62.194.123.84:7759} cannot be found in the request cache, skipping it
WARNING 1520539688.33          server:351   No session accepted this data from 125.68.4.51:23616

**Comment by arvidn** [comment]
i've noticed that these counters can become out-of-sync if there's an error and the torrent recovers from the error via a call to `clear_error()`. Is there some event you see where the file size counters become incorrect? issuing a force-recheck for instance?

**Comment by devos50** [comment]
@arvidn it happens when force-rechecking a torrent.  I followed the exact steps posted by @themusicgod1 to reproduce this.

**Comment by devos50** [comment]
@themusicgod1 yes, the call to libtorrent to fetch file information does not return 100% completion, even though the file is there.

**Comment by themusicgod1** [comment]
upstreamish bug report : https://bugs.launchpad.net/ubuntu/+source/libtorrent-rasterbar/+bug/1774259

**Comment by arvidn** [comment]
after re-reading this issue. There are a few assumptions that are not true in general.
If `1984.pdf` is smaller than a single piece, or if the file does not contain a single whole piece (maybe it only spans two pieces partially) it will *not* contribute to having any more pieces after downloading it.

On the other hand, archive.org tend to align their files with padfiles, which should make it use at least one piece. But perhaps the problem of rechecking is related to padfiles.

Does this reproduce with torrents without padfiles too?
And in this case, how big is 1984.pdf compared to the piece size?
a tool like `dump_torrent` in libtorrent (but probably other tools as well) can print which piece it starts and ends at. if those two pieces are adjacent, there's a risk it doesn't cover a single whole piece.
